velocity_modelling/cvm/velocity3d.py

```diff
@@ -0,0 +1,786 @@
+        Basin membership indicators.
+    logger : VMLogger
+        Logger instance for logging messages.
+    """
+
+    def __init__(self, n_depth: int, logger: VMLogger = None):
+        """
+        Initialize arrays for velocity and density data.
+
+
+        """
+        if logger is None:
+            self.logger = VMLogger(name="velocity_model.qualities_vector")
+        else:
+            self.logger = logger
+
+        self.vp = np.zeros(n_depth, dtype=np.float64)
+        self.vs = np.zeros(n_depth, dtype=np.float64)
+        self.rho = np.zeros(n_depth, dtype=np.float64)
+        self.inbasin = np.zeros(n_depth, dtype=np.int8)
+
+    def log(self, message: str, level: int = None) -> None:
+        """
+        Log a message with the specified level.
+
+        Parameters
+        ----------
+        message : str
+            The message to log.
+        level : int, optional
+            The logging level (default is None).
+        """
+
+        if level is None:
+            level = self.logger.INFO
+        self.logger.log(message, level)
+
+    def prescribe_velocities(
+        self,
+        global_params: dict,
+        velo_mod_1d_data: VelocityModel1D,
+        nz_tomography_data: TomographyData,
+        global_surfaces: GlobalSurfaces,
+        mesh_vector: MeshVector,
+        partial_global_surface_depths: PartialGlobalSurfaceDepths,
+        partial_basin_surface_depths_list: list[PartialBasinSurfaceDepths],
+        in_basin_list: list[InBasin],
+        topo_type: TopoTypes,
+        on_boundary: bool,
+    ):
+        """
+        Calculate velocities and densities for all depths at a given lat-lon point.
+
+        Parameters
+        ----------
+        global_params : dict
+            Parameters for the global velocity model.
+        velo_mod_1d_data : VelocityModel1D
+            1D velocity model data.
+        nz_tomography_data : TomographyData
+            Tomography data for velocity adjustments.
+        global_surfaces : GlobalSurfaces
+            Global surface definitions.
+        mesh_vector : MeshVector
+            Lat-lon point with multiple depths.
+        partial_global_surface_depths : PartialGlobalSurfaceDepths
+            Global surface depth data at this location.
+        partial_basin_surface_depths_list : list[PartialBasinSurfaceDepths]
+            Basin surface depth data at this location.
+        in_basin_list : list[InBasin]
+            Basin membership indicators.
+        topo_type : TopoTypes
+            Topography handling method ("TRUE", "BULLDOZED", "SQUASHED", "SQUASHED_TAPERED").
+        on_boundary : bool
+            Whether this point is on a model boundary.
+        """
+
+        partial_global_surface_depths.interpolate_global_surface_depths(
+            global_surfaces, mesh_vector
+        )
+
+        in_any_basin_lat_lon = any(
+            in_basin.in_basin_lat_lon for in_basin in in_basin_list
+        )
+        # TODO: test this
```

Comment:
Are these test this a note for future or for something that needed to be tested if it works within this PR?

Reply:
I meant they are not thoroughly tested:  but after I opened this PR, I played with the randomized parameter testing that covered this - So I'll remove these comments.